src/pudl/helpers.py

```diff
@@ -2327,3 +2332,152 @@ def standardize_phone_column(df: pd.DataFrame, columns: list[str]) -> pd.DataFra
+        partitions: Partitions which correspond to the table_data. If passed
+            ``{'years': 1995}`` then this method will produce a parquet file at the path
+            ``PudlPaths().parquet_transform_dir / table_name / '1995.parquet'``.
+    """
+    # Create ParquetData class to get path to write parquet file
+    parquet_data = ParquetData(table_name=table_name, partitions=partitions)
+    if isinstance(table_data, pd.DataFrame):
+        table_data.to_parquet(parquet_data.parquet_path)
+    elif isinstance(table_data, pl.LazyFrame):
+        table_data.sink_parquet(parquet_data.parquet_path, engine="streaming")
+    elif isinstance(table_data, duckdb.DuckDBPyRelation):
+        table_data.to_parquet(str(parquet_data.parquet_path), overwrite=True)
+    else:
+        raise TypeError(
+            "table_data must be of type pd.DataFrame, pl.LazyFrame or duckdb.DuckDBPyRelation."
+            f" Got {type(table_data)}."
+        )
+    return parquet_data
+
+
+def lf_from_parquet(
+    parquet_data: ParquetData, use_all_partitions: bool = False
+) -> pl.LazyFrame:
+    """Scan parquet file(s) from disk and return Polars LazyFrame.
+
+    Args:
+        parquet_data: Points to parquet data on disk.
+        use_all_partitions: If true read the entire directory of parquet files.
+            Otherwise only read data from the partition specified in parquet_data.
+    """
+    if use_all_partitions:
+        return pl.scan_parquet(parquet_data.parquet_directory)
+    return pl.scan_parquet(parquet_data.parquet_path)
+
+
+def df_from_parquet(
+    parquet_data: ParquetData, use_all_partitions: bool = False
+) -> pd.DataFrame:
+    """Read data from a set of parquet files and return a pandas DataFrame.
+
+    Args:
+        parquet_data: Points to parquet data on disk.
+        use_all_partitions: If true read the entire directory of parquet files.
+            Otherwise only read data from the partition specified in parquet_data.
+    """
+    if use_all_partitions:
+        return pd.read_parquet(parquet_data.parquet_directory)
+    return pd.read_parquet(parquet_data.parquet_path)
+
+
+@contextmanager
+def duckdb_relation_from_parquet(
+    parquet_data: ParquetData, use_all_partitions: bool = False
+) -> duckdb.DuckDBPyRelation:
+    """Create a duckdb relation to read from parquet files.
+
+    This method is intended to be used as a context manager to keep the duckdb
+    connection open while the relation is in use.
+
+    Args:
+        parquet_data: Points to parquet data on disk.
+        use_all_partitions: If true read the entire directory of parquet files.
+            Otherwise only read data from the partition specified in parquet_data.
+    """
+    with duckdb.connect() as conn:
+        if use_all_partitions:
+            yield conn.read_parquet(f"{parquet_data.parquet_directory}/*.parquet")
+        yield conn.read_parquet(str(parquet_data.parquet_path))
+
+
+def duckdb_extract_zipped_csv(
```

Comment:
This is a new simplified helper function for extracting CSVs from zips. I think it has a constrained enough amount of functionality to be useful in other cases, but could also move this the `pudl.extract.vcerare` if it still feels too specific.

Reply:
This function is pretty tightly coupled to the other helper functions I added here (which aren't all related to CSV extraction), and I think it would be confusing to split them up.